docs/shared/guides/unknown-local-cache.md

```diff
@@ -0,0 +1,72 @@
+# Unknown Local Cache Error
+
+This document will explain why the following error happens and how to address it.
+
+```
+NX Invalid Cache Directory for Task "myapp:build"
+
+The local cache artifact in "node_modules/.cache/nx/786524780459028195" has not been generated on this machine.
+As a result, the cache's content integrity cannot be confirmed, which may make cache restoration potentially unsafe.
+If your machine ID has changed since the artifact was cached, run "nx reset" to fix this issue.
+Read about the error and how to address it here: https://nx.dev/recipes/other/unknown-local-cache
+```
+
+## Nx Tracks Cache Source
+
+Nx can cache tasks, which can drastically speed up your CI and local builds. However, this comes with the potential risk
+of "cache poisoning", where cache artifacts could be intentionally or inadvertently overwritten. If another user
+executes a task that matches the hash of the tainted artifact, they could retrieve the corrupted artifact. Nx and Nx
+Cloud contain several safeguards to minimize the likelihood of cache poisoning or, in the case of Nx Cloud, completely
+prevent it.
+
+The error above is one such safeguard.
+
+Nx trusts the local cache. If you executed a task and stored the corresponding cached artifact on your machine, you can
+safely restore it on the same machine without worrying about cache poisoning. The "Invalid Cache Directory" error is
+thrown when the local cache artifact was created on another machine.
```

Comment:
```suggestion
of "cache poisoning", where cache artifacts could be intentionally or inadvertently overwritten. When Nx
executes a task that matches the hash of the tainted artifact, it would retrieve the corrupted artifact and use it as the outcome of the task.

The above error is a safeguard to mitigate the unintentionally usage of a cache artifact from the local cache which may be unsafe.

Nx expects that artifacts in local cache on a machine are created by the machine itself. When this is the case, in order to tamper with the cache artifact, the actor would need access to the machine itself. However, when artifacts in the local cache are created by a different machine, there is reason for concern. By default, Nx will refuse to use such artifacts as they may have been poisoned. The "Invalid Cache Directory" error is
thrown when the local cache artifact was created on another machine.
```